docs/change-log.md

```diff
@@ -20,6 +20,7 @@
 
 *Note for later: release as semver-minor.*
 
+- Reduced download size (npm package does not include archive of previous docs anymore)
```

Comment:
Just to match the other lines in style and format. Also, the bit about why could be made a little more concise and precise.

```suggestion
- Reduced package download size substantially by removing the archive of previous releases' documentation. ([#2561](https://github.com/MithrilJS/mithril.js/pull/2561) [@cztomsik](https://github.com/cztomsik))
```